mg/builtins.go

```diff
@@ -92,14 +103,17 @@ func (bc BuiltinCmds) EnvCmd(bx *BultinCmdCtx) *State {
 	return bx.State
 }
 
+// Commands returns a list of predefined commands.
 func (bc BuiltinCmds) Commands() BultinCmdList {
 	return []BultinCmd{
-		BultinCmd{Name: ".env", Desc: "List env vars", Run: bc.EnvCmd},
+		BultinCmd{Name: ".env", Desc: "List env vars", Run: EnvCmd},
 		BultinCmd{Name: ".exec", Desc: "Run a command through os/exec", Run: bc.ExecCmd},
-		BultinCmd{Name: ".type", Desc: "Lists all builtins or which builtin handles a command", Run: bc.TypeCmd},
+		BultinCmd{Name: ".type", Desc: "Lists all builtins or which builtin handles a command", Run: TypeCmd},
 	}
 }
 
+// Reduce will return a copy of the mx.State while adding the predefined
+// commands to its builtins, if the mx.Action is a RunCmd.
 func (bc BuiltinCmds) Reduce(mx *Ctx) *State {
```

Comment:
I'd wprobably ord this like `Reduce adds the list of predefined builtins for the RunCmd action`. It's already implied that if the state changes, the reducer returns a new state object.